SwiftSampleProject/SwiftyVokoderTests/ManagedObjectContextTests.swift

```diff
@@ -0,0 +1,156 @@
+     
+        let newCount = self.manager.countForClass(TrainLine.self)
+        XCTAssertEqual(newCount, countOfTrainLines + 1)
+    }
+    
+    func testSaveWithoutWaitingEventuallySaves() {
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+
+        self.manager.deleteAllObjectsOfClass(Station.self, context: nil)
+        
+        self.expectationForNotification(NSManagedObjectContextDidSaveNotification,
+            object: self.manager.managedObjectContext) { _ in
+                
+                let newCountOfStations = self.manager.countForClass(Station.self)
+                XCTAssertEqual(newCountOfStations, 0)
+                XCTAssertNotEqual(countOfStations, newCountOfStations)
+                
+                return true
+        }
+        
+        guard let rootContext = self.manager.managedObjectContext.parentContext else {
+            XCTFail("Expecting the main context to have a parent context")
+            return
+        }
+        self.expectationForNotification(NSManagedObjectContextDidSaveNotification,
+            object: rootContext) { _ in
+                
+                let newCountOfStations = self.manager.countForClass(Station.self, forContext: rootContext)
+                XCTAssertEqual(newCountOfStations, 0)
+                XCTAssertNotEqual(countOfStations, newCountOfStations)
+                
+                return true
+        }
+    
+        self.manager.saveMainContext()
+        
+        self.waitForExpectationsWithTimeout(10, handler: nil)
+    }
+    
+    func testSaveAndMergeWithMainContextSavesGrandChildren() {
+        let childContext = self.manager.temporaryContext()
+        let grandChildContext = NSManagedObjectContext(concurrencyType: .PrivateQueueConcurrencyType)
+        grandChildContext.parentContext = childContext
+        
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+        
+        grandChildContext.performBlockAndWait {
+            self.manager.deleteAllObjectsOfClass(Station.self, context: grandChildContext)
+        }
+        
+        self.manager.saveAndMergeWithMainContextAndWait(grandChildContext)
+        
+        let newCountOfStations = self.manager.countForClass(Station.self)
+        XCTAssertEqual(newCountOfStations, 0)
+        XCTAssertNotEqual(countOfStations, newCountOfStations)
+    }
+    
+    func testUnsavedMainContextChangesGetPassedToTempContexts() {
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+
+        let childContextBeforeChanges = self.manager.temporaryContext()
+        self.manager.deleteAllObjectsOfClass(Station.self, context: nil)
+        let childContextAfterChanges = self.manager.temporaryContext()
+
+        let childCountOfStations = self.manager.countForClass(Station.self, forContext: childContextBeforeChanges)
```

Comment:
can you add a bit of documentation about exactly what this is testing - it took me a second to figure out that the two `temporaryContext` calls wouldn't be pointed at the same instance